canvas/api/assignment/fetchsubmissionfiles.py

```diff
@@ -100,25 +102,41 @@ def _write_files(temp_dir, file_contents):
     return len(file_contents)
 
 def _download_files(temp_dir, file_targets):
+    num_files = sum([len(targets) for targets in file_targets.values()])
+    if (num_files == 0):
+        return 0
+
     count = 0
+    max_workers = min(MIN_WORKERS , num_files)
+    tasks = []
 
-    for (email, targets) in file_targets.items():
+    for email, targets in file_targets.items():
         user_dir = os.path.join(temp_dir, email)
         os.makedirs(user_dir, exist_ok = True)
 
-        for (filename, url) in targets:
-            path = os.path.join(user_dir, filename)
+        for filename, url in targets:
+            tasks.append((email, filename, url, user_dir))
+
+    try:
```

Comment:
This try seems misplaced.
You are not catching individual errors, i.e., one error will stop the entire process.